packages/server/src/controllers/user.ts

```diff
@@ -0,0 +1,189 @@
+//  Required Libraries
+import jwt from 'jsonwebtoken';
+import axios from 'axios';
+import * as crypto from 'crypto';
+//  Import types
+import type { Request, Response } from 'express';
+import type {
+  AuthRequest,
+  MailOptions, RequestWithUserID,
+} from '../types.js';
+// Import services
+import transport from '../services/createTransport.js';
+//  DB Model
+import User from '../models/user.js';
+import OtpModel from '../models/otp.js';
+
+//  Register User
+export const registerUser = async (req: Request, res: Response) => {
+  try {
+    const { email } = req.body;
+    const existingUser = await User.findOne({ email });
+    if (existingUser) {
+      return res.status(400).json({ mesage: 'User already Exists!' });
+    }
+    // Generate a random 6-digit OTP. Generation to be updated
+    const otpCode = crypto.randomInt(100000, 999999);
+    // Save OTP in a new collection. To be implemented
+    const otp = new OtpModel({
+      email,
+      code: otpCode,
+    });
+    await otp.save();
+    //  Create user
+    const user = new User({
+      email,
+      password: null,
+      files: null,
+      isVerified: false,
+      secret: null,
+    });
+    await user.save();
+
+    //  Create data token
+    const dataToken = jwt.sign({ userID: email, isValid: false, role: 'user' }, process.env.JWT_SECRET as string);
+
+    // Send the OTP to the user's email
+    const mailOptions: MailOptions = {
+      // Enter a Mail if nothing in environment variables
+      from: process.env.EMAIL || '',
+      to: email,
+      subject: 'Your One time password is here! ',
+      // text: 'Hello' ,//  change as required
+      html: `<b>Hello User your OTP is: ${otpCode}.</b><br>The OTP will expired in 5 minutes.`,
+    };
+    try {
+      await transport.sendMail(mailOptions);
+      return res.status(250).json({ message: 'Verification email sent! Please check your inbox.', token: dataToken });
+    } catch (err) {
+      return res.status(500).json({ message: 'Error sending verification email. Please try again in sometime.' });
+    }
+  } catch (err) {
```

Comment:
The first catch block is to handle any error thrown if nodemailer is not able to send a mail to the users account. The second one is to handle errors related to the route in case when the user makes a request and is not recieving a response. For example an error could be thrown due to the DB connectivity. To test this remove the email id and password in the .env file.